edx_rest_framework_extensions/authentication.py

```diff
@@ -148,3 +149,71 @@ def process_user_info_response(self, response):
 
     def authenticate_header(self, request):
         return 'Bearer'
+
+
+class JwtAuthentication(JSONWebTokenAuthentication):
+    """Get or create the user corresponding to the provided JWT.
+
+    Clients should authenticate themselves by passing a JWT in the Authorization
+    HTTP header, prepended with the string 'JWT'.
+
+    At a minimum, the JWT payload must contain a username. If an email address
+    is provided in the payload, it will be used to update the retrieved user's
+    email address which surfaces in the Oscar management interface.
+
+    Additionally, clients may enclose a 'tracking_context' sub-dictionary in the
+    JWT payload.  If present, the service will update itself with the value
+    of the tracking_context, and will include its contents when firing tracking
+    events to analytics services (if and when configured to do so).
+
+    Example:
+        Access an endpoint protected by JWT authentication as follows.
+
+        >>> url = 'http://localhost:8002/api/v2/baskets/'  # Protected by JwtAuthentication
+        >>> token = jwt.encode({'username': 'Saul', 'email': 'saul@bettercallsaul.com'}, 'insecure-secret-key')
+        >>> headers = {
+            'content-type': 'application/json',
+            'Authorization': 'JWT ' + token
+        }
+        >>> data = {'products': [{'sku': 'PAID-SEAT'}], 'checkout': True, 'payment_processor_name': 'paypal'}
+        >>> response = requests.post(url, data=json.dumps(data), headers=headers)
+        >>> response.status_code
+        200
+    """
+    def _get_jwt_payload_attributes(self):
+        """ Returns the JWT payload attributes which should be used to update the local user model. """
+        return get_setting('JWT_PAYLOAD_ATTRIBUTES')
+
+    def authenticate(self, request):
+        try:
+            return super(JwtAuthentication, self).authenticate(request)
+        except Exception as ex:
+            # Errors in production do not need to be logged (as they may be noisy),
+            # but debug logging can help quickly resolve issues during development.
+            logger.debug(ex)
+            raise
+
+    def authenticate_credentials(self, payload):
+        """Get or create an active user with the username contained in the payload."""
+        username = payload.get('preferred_username') or payload.get('username')
+
+        if username is None:
+            raise exceptions.AuthenticationFailed('Invalid payload.')
+        else:
+            try:
+                user, __ = User.objects.get_or_create(username=username)
+                is_update = False
```

Comment:
I always find this variable name confusing with the call above. Would you rename it to `attributes_updated`, please?